### PR TITLE
Make function parameter names consistent

### PR DESCRIPTION
**Pull request by mkindahl**
Renaming the parameter `hypertable_or_cagg` in functions `drop_chunks`
and `show_chunks` to `relation` and changing parameter name from
`main_table` to `hypertable` or `relation` depending on context.

### CODE REVIEW

File: sql/size_utils.sql [hunk 1]
```diff
@@ -367,7 +367,7 @@ $BODY$;
 
 -- Convenience function to return approximate row count
 --
--- main_table - table or hypertable to get approximate row count for
+-- hypertable - table or hypertable to get approximate row count for
```
Comment (gayyappan) [inline sql/size_utils.sql:370]: i think this works for regular PG tables too. (cc @pmwkaa ). so shouldn't use hypertable here. Functions that do not have hypertable/chunks prefix work for other PG objects as well.

File: sql/ddl_api.sql [hunk 2]
```diff
@@ -23,7 +23,7 @@
 -- replication_factor - (Optional) A value of 1 or greater makes this hypertable distributed
 -- data_nodes - (Optional) The specific data nodes to distribute this hypertable across
 CREATE OR REPLACE FUNCTION  create_hypertable(
-    main_table              REGCLASS,
+    hypertable              REGCLASS,
```
Comment (erimatnor) [inline sql/ddl_api.sql:26]: This would be `relation` since there is no hypertable at this point.
```suggestion
    relation              REGCLASS,
```

File: sql/size_utils.sql (hunk 1 above)
Comment (gayyappan) [reply sql/size_utils.sql:370]: oh see that the change is only to the comments. so just need to fix the comment.

File: sql/ddl_api.sql [hunk 3]
```diff
@@ -43,7 +43,7 @@ CREATE OR REPLACE FUNCTION  create_hypertable(
 
 -- Same functionality as create_hypertable, only must have a replication factor > 0 (defaults to 1)
 CREATE OR REPLACE FUNCTION  create_distributed_hypertable(
-    main_table              REGCLASS,
+    hypertable              REGCLASS,
```
Comment (erimatnor) [inline sql/ddl_api.sql:46]: This would be `relation` since there is no hypertable at this point.
```suggestion
    relation              REGCLASS,
```

**Review by erimatnor — COMMENTED** [review]
I suspect there are some update script changes missing. Also, wasn't there some policy functions that needed a change?

**Comment by erimatnor** [comment]
Nit: the PR commit title is confusing. It sounds as if "parameters" are renamed to "functions".

I suggest changing "Make parameter names consistent" or something

File: sql/size_utils.sql (hunk 1 above)
Comment (mkindahl) [reply sql/size_utils.sql]: Yes, but I think the parameter name is changed in the SQL file, so we should use `relation` here.

File: sql/ddl_api.sql [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 -- Converts a regular postgres table to a hypertable.
 --
--- main_table - The OID of the table to be converted
+-- hypertable - The OID of the table to be converted
```
Comment (erimatnor) [inline sql/ddl_api.sql:9]:
```suggestion
-- relation - The OID of the table to be converted
```

File: src/hypertable.c [hunk 5]
```diff
@@ -1761,7 +1761,7 @@ TS_FUNCTION_INFO_V1(ts_hypertable_distributed_create);
  * Create a hypertable from an existing table.
  *
  * Arguments:
- * main_table              REGCLASS
+ * hypertable              REGCLASS
```
Comment (erimatnor) [inline src/hypertable.c:1764]:
```suggestion
 * relation              REGCLASS
```

**Review by erimatnor — COMMENTED** [review]
I would still prefer a less confusing PR/commit title. I assume the commits will be cleaned up anyway.

**Comment by mkindahl** [comment]
> Nit: the PR commit title is confusing. It sounds as if "parameters" are renamed to "functions".
> 
> I suggest changing "Make function parameter names consistent" or something

I'll squash and update the final commit before pushing.

**Comment by mkindahl** [comment]
> I would still prefer a less confusing PR/commit title. I assume the commits will be cleaned up anyway. Still missing update script.

Renamed the commit.

File: sql/updates/latest-dev.sql [hunk 6]
```diff
@@ -17,13 +20,17 @@ DROP VIEW IF EXISTS timescaledb_information.license;
 
 -- Add new function definitions, columns and tables for distributed hypertables
 DROP FUNCTION IF EXISTS create_hypertable(regclass,name,name,integer,name,name,anyelement,boolean,boolean,regproc,boolean,text,regproc,regproc);
+DROP FUNCTION IF EXISTS create_distributed_hypertable(regclass,name,name,integer,name,name,anyelement,boolean,boolean,regproc,boolean,text,regproc,regproc,integer,name[]);
```
Comment (erimatnor) [inline sql/updates/latest-dev.sql:23]: Is this line really needed given that `create_distribute_hypertable` didn't exist in a previous version?

File: sql/updates/latest-dev.sql [hunk 7]
```diff
@@ -1,6 +1,7 @@
 --Drop functions in size_utils and dependencies, ordering matters.
 -- Do not reorder
 DROP VIEW IF EXISTS timescaledb_information.hypertable;
+DROP FUNCTION IF EXISTS hypertable_detailed_size;
```
Comment (erimatnor) [inline sql/updates/latest-dev.sql:4]: Isn't this a new function? Thus nothing to remove here.

File: sql/updates/latest-dev.sql [hunk 8]
```diff
@@ -9,6 +10,8 @@ DROP FUNCTION IF EXISTS indexes_relation_size_pretty;
 DROP FUNCTION IF EXISTS indexes_relation_size;
 DROP FUNCTION IF EXISTS _timescaledb_internal.partitioning_column_to_pretty;
 DROP FUNCTION IF EXISTS _timescaledb_internal.range_value_to_pretty;
+DROP FUNCTION IF EXISTS hypertable_compression_stats;
+DROP FUNCTION IF EXISTS chunk_compression_stats;
```
Comment (erimatnor) [inline sql/updates/latest-dev.sql:14]: I think these two functions are also new for this version.

**Comment by mkindahl** [comment]
> I suspect there are some update script changes missing. Also, wasn't there some policy functions that needed a change?

The policies all use either `continuous_aggregate` or `hypertable` as parameter names. I assumed that these should not be changed.

I have added the removal of some functions in the update since I am not entirely sure what entails a "previous version" in this context.

File: sql/updates/latest-dev.sql (hunk 6 above)
Comment (mkindahl) [reply sql/updates/latest-dev.sql:23]: I am unsure what "previous version" means in this context. The previous beta or 1.7? I can test to remove it and see if the update script works.

**Comment by k-rus** [comment]
> The policies all use either `continuous_aggregate` or `hypertable` as parameter names. I assumed that these should not be changed.

Reorder and compression policies are only applied to hypertables. Continuous aggregate policy is only applied to continuous aggregate, which is a view. Retention policy can be applied to both hypertables and continuous aggregates.

**Comment by mkindahl** [comment]
> > The policies all use either `continuous_aggregate` or `hypertable` as parameter names. I assumed that these should not be changed.
> 
> @mkindahl Reorder and compression policies are only applied to hypertables. Continuous aggregate policy is only applied to continuous aggregate, which is a view. Retention policy can be applied to both hypertables and continuous aggregates.

Thanks @k-rus . Changed retention policy functions to use `relation` instead.

File: sql/ddl_api.sql [hunk 9]
```diff
@@ -102,22 +102,22 @@ LANGUAGE C VOLATILE PARALLEL UNSAFE;
 -- show chunks older than or newer than a specific time.
 -- `hypertable` must be a valid hypertable or continuous aggregate.
```
Comment (k-rus) [inline sql/ddl_api.sql:103]:
```suggestion
-- `relation` must be a valid hypertable or continuous aggregate.
```

File: sql/size_utils.sql (hunk 1 above)
Comment (pmwkaa) [reply sql/size_utils.sql]: Relation sounds good to me as well

**Review by pmwkaa — APPROVED** [review]
Looks good to me

File: tsl/test/expected/continuous_aggs_policy.out [hunk 10]
```diff
@@ -163,6 +163,7 @@ CREATE MATERIALIZED VIEW max_mat_view_timestamp
         GROUP BY 1 WITH NO DATA;
 SELECT add_continuous_aggregate_policy('max_mat_view_timestamp', '10 day', '1 h'::interval , '1 h'::interval) as job_id \gset
 CALL run_job(:job_id);
+NOTICE:  continuous aggregate "max_mat_view_timestamp" is already up-to-date
```
Comment (k-rus) [inline tsl/test/expected/continuous_aggs_policy.out:166]: Unrelated change to the PR. IMHO it should be handled separately, since it is unclear what the reason for this sudden problem.

Comment (erimatnor) [inline tsl/test/expected/continuous_aggs_policy.out:166]: This is an unexpected change in the test.

**Review by k-rus — CHANGES_REQUESTED** [review]
This PR contains a change, which is not related to this PR. I don't understand why the unrelated test suddenly started to fail and thus suggest to handle in a separate PR, so it is not masked here.

I found two places where changes are still needed.

File: sql/updates/latest-dev.sql (hunk 6 above)
Comment (erimatnor) [reply sql/updates/latest-dev.sql]: Betas are not upgradable, so the most recent version we can upgrade from is 1.7.4.

File: tsl/test/expected/continuous_aggs_policy.out (hunk 10 above)
Comment (mkindahl) [reply tsl/test/expected/continuous_aggs_policy.out:166]: Tests won't pass if this line is not here so how can it be handled separately?

Comment (mkindahl) [reply tsl/test/expected/continuous_aggs_policy.out:166]: Yes, it is, but tests won't pass if I do not add this line.

Comment (k-rus) [reply tsl/test/expected/continuous_aggs_policy.out:166]: It is really weird that the notice disappeared from the test.
handling separately: https://github.com/timescale/timescaledb/pull/2475

Comment (mkindahl) [reply tsl/test/expected/continuous_aggs_policy.out:166]: Ok. Will wait for you to push that one then.

Comment (mkindahl) [reply tsl/test/expected/continuous_aggs_policy.out:166]: Seems you closed that issue without pushing, so I should probably push this even though it isn't related to the change.

File: sql/updates/latest-dev.sql (hunk 6 above)
Comment (mkindahl) [reply sql/updates/latest-dev.sql]: Update CI worked without these lines, so removed them.

File: tsl/test/expected/continuous_aggs_policy.out (hunk 10 above)
Comment (k-rus) [reply tsl/test/expected/continuous_aggs_policy.out:166]: I block your PR, this line should not be pushed. The notice will likely disappear tomorrow, since it is a flaky test. See https://github.com/timescale/timescaledb/pull/2477 as a temporal solution for now.

Comment (mkindahl) [reply tsl/test/expected/continuous_aggs_policy.out:166]: Then I'll be waiting for you to say when I can remove this line.

**Review by k-rus — CHANGES_REQUESTED** [review]
The unrelated change needs to be removed from the PR.

**Review by k-rus — APPROVED** [review]

**Review by svenklemm — APPROVED** [review]

**Review by WireBaron — APPROVED** [review]
